src/main/java/forer/ApprovedDrugs/DrugController.java

```diff
@@ -0,0 +1,71 @@
+package forer.ApprovedDrugs;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.net.URL;
+import javax.swing.JLabel;
+import com.google.inject.Inject;
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+public class DrugController {
+
+    private DrugView view;
+    private DrugService service;
+    DrugFeed feed;
+    private Disposable disposable;
+
+    @Inject
+    public DrugController(DrugService service, DrugView view) {
+        this.view = view;
+        this.service = service;
+    }
+
+    public void requestDrugFeed() {
+            disposable = Observable.interval(0,10000, TimeUnit.SECONDS)
+                    .flatMap(aLong -> service.getApprovedDrugs())
+                    .map(DrugFeed::getMolecules)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(Schedulers.single())
+                    .subscribe(this :: setMolecules,
+                               throwable -> System.out.println("Error getting data"));
```

Comment:
Always,
ALWAYS.
Print the *exact* error you're getting.
In this case, instead of 
`throwable -> System.out.println("Error getting data")`
do
`throwable -> throwable.printStackTrace()`

Reply:
Then tell me what your Exception is.